OsmAnd/src/net/osmand/plus/measurementtool/adapter/MeasurementToolAdapter.java

```diff
@@ -88,15 +90,27 @@ public boolean onTouch(View view, MotionEvent event) {
 		if (!TextUtils.isEmpty(pointDesc)) {
 			holder.descr.setText(pointDesc);
 		} else {
+			String azimuth;
+			String text = "";
 			if (pos < 1) {
-				holder.descr.setText(mapActivity.getString(R.string.shared_string_control_start));
+				text = mapActivity.getString(R.string.shared_string_control_start);
+				Location l1 = mapActivity.getMyApplication().getLocationProvider().getLastStaleKnownLocation();
+				if (mapActivity.getMyApplication().getLocationProvider().getLastKnownLocation() != null) {
+					azimuth = OsmAndFormatter.getFormattedAzimuth(MapUtils.getBearingToPoint(l1, points.get(0).lat, points.get(0).lon));
+					text = text + BULLET + azimuth;
+				}
+				holder.descr.setText(text);
 			} else {
 				float dist = 0;
 				for (int i = 1; i <= pos; i++) {
 					dist += MapUtils.getDistance(points.get(i - 1).lat, points.get(i - 1).lon,
 							points.get(i).lat, points.get(i).lon);
```

Comment:
distance and bearing could be measured in 1 call  l1.bearingTo(l2);  l1.distTo(l2);

Reply:
Dunno how it will help, but added one more method to MapUtils which returns distance and bearing.